src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -2407,7 +2407,7 @@ public unsafe Object GetValue(params int[] indices)
         private unsafe Object GetValue(int* pIndices, int rank)
         {
             if (this.Rank != rank)
-                throw new ArgumentException(SR.Arg_RankIndices);
+                throw new ArgumentException(SR.Arg_RankIndices, nameof(rank));
```

Comment:
This is a helper method that is validating the arguments of the actual public methods. The actual public methods do not have rank argument. The `rank` argument name won't make sense if somebody gets this exception. The rank argument  it is just an internal implementation detail that should not leak to the exception message.